drivers/clock_control/nrf_power_clock.c

```diff
@@ -226,68 +220,277 @@ static inline void anomaly_132_workaround(void)
 #endif
 }
 
-static int clock_async_start(struct device *dev,
-			     clock_control_subsys_t subsys,
-			     struct clock_control_async_data *data)
+static void lfclk_start(void)
 {
-	enum clock_control_nrf_type type = (enum clock_control_nrf_type)subsys;
-	const struct nrf_clock_control_sub_config *config;
-	struct nrf_clock_control_sub_data *clk_data;
-	int key;
-	uint8_t ref;
 
-	__ASSERT_NO_MSG(type < CLOCK_CONTROL_NRF_TYPE_COUNT);
-	config = get_sub_config(dev, type);
-	clk_data = get_sub_data(dev, type);
-
-	__ASSERT_NO_MSG((data == NULL) ||
-			((data != NULL) && (data->cb != NULL)));
-
-	/* if node is in the list it means that it is scheduled for
-	 * the second time.
-	 */
-	if ((data != NULL)
-	    && is_in_list(&clk_data->list, &data->node)) {
-		return -EBUSY;
+	if (IS_ENABLED(CONFIG_NRF52_ANOMALY_132_WORKAROUND)) {
+		anomaly_132_workaround();
 	}
 
-	key = irq_lock();
-	ref = ++clk_data->ref;
-	__ASSERT_NO_MSG(clk_data->ref > 0);
-	irq_unlock(key);
+	nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_LFCLKSTART);
+}
 
-	if (data) {
-		bool already_started;
+static void lfclk_stop(void)
+{
+	if (IS_ENABLED(CONFIG_CLOCK_CONTROL_NRF_K32SRC_RC_CALIBRATION)) {
+		z_nrf_clock_calibration_lfclk_stopped();
+	}
 
-		clock_irqs_disable();
-		already_started = clk_data->started;
-		if (!already_started) {
-			list_append(&clk_data->list, &data->node);
-		}
-		clock_irqs_enable();
+	nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_LFCLKSTOP);
+}
+
+static void hfclk_start(void)
+{
+	nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_HFCLKSTART);
+}
+
+static void hfclk_stop(void)
+{
+	nrf_clock_task_trigger(NRF_CLOCK, NRF_CLOCK_TASK_HFCLKSTOP);
+}
+
+static uint32_t *get_hf_flags(void)
+{
+	struct nrf_clock_control_data *data =
+					DEVICE_GET(clock_nrf)->driver_data;
 
-		if (already_started) {
-			data->cb(dev, subsys, data->user_data);
+	return &data->subsys[CLOCK_CONTROL_NRF_TYPE_HFCLK].flags;
+}
+
+static void generic_hfclk_start(void)
+{
+	atomic_t *flags = get_hf_flags();
+	nrf_clock_hfclk_t type;
+	bool already_started = false;
+	int key = irq_lock();
+
+	*flags |= HF_GENERIC_ACTIVE;
+	if (*flags & HF_BT_ACTIVE) {
+		nrf_clock_is_running(NRF_CLOCK, NRF_CLOCK_DOMAIN_HFCLK, &type);
```

Comment:
The value returned by `nrf_clock_is_running()` is ignored here, so `(void)` could help make it clear (and perhaps also avoid warnings from some linters).